src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php

```diff
@@ -30,18 +31,27 @@ abstract class AbstractFailedMessagesCommand extends Command
 {
     private $receiverName;
     private $receiver;
+    /**
+     * @var ServiceLocator
+     */
+    private $failureTransports;
 
-    public function __construct(string $receiverName, ReceiverInterface $receiver)
+    public function __construct(?string $receiverName, ?ReceiverInterface $receiver, ServiceLocator $failureTransports)
     {
         $this->receiverName = $receiverName;
         $this->receiver = $receiver;
+        $this->failureTransports = $failureTransports;
 
         parent::__construct();
     }
 
-    protected function getReceiverName(): string
+    protected function getReceiverName(?string $name): string
```

Comment:
Also, for the case of (B) (where we are using the global/default transport because no option was passed), we should print a message telling the user how to use different transports (if there are any):

> To use a different failure transport, re-run the command with --failure-transport=
> Available failure transports are: foo, baz, larry